weakmap.c

```diff
@@ -398,7 +432,33 @@ wmap_aset_replace(st_data_t *key, st_data_t *val, st_data_t new_key_ptr, int exi
     return ST_CONTINUE;
 }
 
-/* Creates a weak reference from the given key to the given value */
+/*
+ *  call-seq:
+ *    map[key] = value -> value
+ *
+ *  Associates the given +value+ with the given +key+; returns +value+.
+ *
+ *  The reference to both +key+ and +value+ are weak, so when there is no
```

Comment:
I think this should be class level documentation, not documentation specific to the `[]=` method, as it describes how the class works in general.